### ISSUE TITLE
Late definition of sort not accepted in mutual block

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#6650]
When trying to build a test for `univSort`, I noticed that this code is rejected by Agda (since forever, I suppose):
```agda
module _ where

  X : Set₁

  postulate
    A : X
    a : A  -- error here

  X = Set
```
```
X should be a sort, but it isn't
when checking that the expression A has type _1
```
Since `X` is blocked by its missing definition, Agda considers it not to be a sort.

It works if `X` resolves to a meta, and it even solves `X` then:
```agda

module _ where

  mutual-block : Set₁

  X : Set₁
  X = _

  postulate
    A : X
    a : A

  mutual-block = Set
```
This is a bit reminiscent of #1113.

Relevant code could be this one:
https://github.com/agda/agda/blob/4a1057bfd5b990cf1468755e67c3b88ec93d8cc8/src/full/Agda/TypeChecking/Sort.hs#L149-L158
Note that we only postpone if we are blocked by meta (`Blocked`), not by e.g. `NotBlocked MissingClauses` which would be relevant here.

**Comment by nad** [comment]
> When trying to build a test for `univSort`, I noticed that this code is rejected by Agda (since forever, I suppose):

I think this is fine. Can you give a good reason for accepting the code?

**Comment by andreasabel** [comment]
Maybe it is fine actually.  The error could be more precise.  Instead
```
X should be a sort, but it isn't
```
it could be
```
X should be a sort, but it isn't known to be one
```
But let us not worry about these details...
